extension/src/helpers/stellar.ts

```diff
@@ -42,8 +42,10 @@ export const getTransactionInfo = (search: string) => {
     (operation: { type: string }) => operation.type,
   );
 
+  const _accountToSign = accountToSign && accountToSign.length > 0 ? accountToSign : transaction._source
```

Comment:
A secondary issue is that `transaction._source` could be any public key, even one not available to Freighter. That will trigger the UI to block signing [here](https://github.com/stellar/freighter/blob/master/extension/src/popup/views/SignTransaction/index.tsx#L356)

Reply:
`In the absence of accountToSign, when it gets [here](https://github.com/stellar/freighter/blob/master/extension/src/popup/views/SignTransaction/index.tsx#L217), it will just use the current active public key`

It seems to me that this is part of the original problem. If I build a tx in the lab for example, and I build that tx to be from a source account which is not one that is available in Freighter, then it will use the current active key without warning the user that they are about to sign a tx that will fail submission because the signer != the source account.

`A secondary issue is that transaction._source could be any public key, even one not available to Freighter. That will trigger the UI to block signing [here](https://github.com/stellar/freighter/blob/master/extension/src/popup/views/SignTransaction/index.tsx#L356)`

Maybe I don't totally understand the issue but isn't that the expected behavior. If the source account is not one in Freighter would we not expect the UI to show the "key not available" info block?